### PR TITLE
[sql lab] simplify the visualize flow

### PR DESCRIPTION
**Pull request by mistercrunch**
The "visualize flow" linking SQL Lab to the "explore view" has never worked so great for people, the current flow is heavy and unintuitive. Now that we have the `MetricsControl` there's less need to define the metrics upfront.

This new flow sends the user straight to the explore view using the table view with all columns (not grouped)

screen shot 2018-07-30 at 12 47 14 pm
screen shot 2018-07-30 at 12 47 36 pm
screen shot 2018-07-30 at 12 47 22 pm

### CODE REVIEW

File: superset/assets/src/SqlLab/components/ExploreResultsButton.jsx [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+  query: {},
+};
+
+class ExploreResultsButton extends React.PureComponent {
+  constructor(props) {
+    super(props);
+    this.state = {
+      hints: [],
+    };
+    this.visualize = this.visualize.bind(this);
```
Comment (hughhhh) [inline superset/assets/src/SqlLab/components/ExploreResultsButton.jsx]: nit: you can use autobind-decorators here

https://github.com/andreypopp/autobind-decorator

File: superset/assets/src/SqlLab/components/ExploreResultsButton.jsx [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+    const uniqueId = shortid.generate();
+    let datasourceName = uniqueId;
+    if (query) {
+      datasourceName = query.user ? `${query.user}-` : '';
+      datasourceName += `${query.tab}-${uniqueId}`;
+    }
+    return datasourceName;
+  }
+  buildVizOptions() {
+    const { schema, sql, dbId, templateParams } = this.props.query;
```
Comment (hughhhh) [inline superset/assets/src/SqlLab/components/ExploreResultsButton.jsx]: nit: 
```
return {
    ...this.props.query,
      datasourceName: this.datasourceName(),
      columns: this.getColumns(),
}
```

File: superset/views/core.py [hunk 3]
```diff
@@ -2201,8 +2202,9 @@ def sync_druid_source(self):
     def sqllab_viz(self):
         SqlaTable = ConnectorRegistry.sources['table']
         data = json.loads(request.form.get('data'))
+        from pprint import pprint
+        pprint(data)
```
Comment (hughhhh) [inline superset/views/core.py]: Remove pprint

File: tests/sqllab_tests.py [hunk 4]
```diff
@@ -236,21 +236,18 @@ def test_sqllab_viz(self):
...
-            },
+            'columns': [{
+                'is_date': False,
+                'type': 'STRING',
+                'nam:qe': 'viz_type',
+                'is_dim': True,
+            }, {
+                'is_date': False,
+                'type': 'OBJECT',
+                'name': 'ccount',
```
Comment (hughhhh) [inline tests/sqllab_tests.py]: `count`

**Comment by betodealmeida** [comment]
Nice, I really like the touch of the timeout dialog.

File: superset/assets/src/SqlLab/components/ExploreResultsButton.jsx (hunk 1 above)
Comment (mistercrunch) [reply superset/assets/src/SqlLab/components/ExploreResultsButton.jsx]: We should introduce this decorator and attack the whole repo with it. In the meantime we have one less dependency and consistent-ish code (we have some binding done outside the constructors in many places...)

File: superset/views/core.py (hunk 3 above)
Comment (mistercrunch) [reply superset/views/core.py]: oops

File: superset/assets/src/SqlLab/components/ExploreResultsButton.jsx (hunk 2 above)
Comment (mistercrunch) [reply superset/assets/src/SqlLab/components/ExploreResultsButton.jsx]: There are lots of other keys in the `query` object we may not want to send over though :(

**Comment by mistercrunch** [comment]
@betodealmeida @hughhhh I addressed most comments

**Review by betodealmeida — APPROVED** [review]
Nice!

**Pull request merged**